src/com/company/exceptions/CheckIndexException.java

```diff
@@ -0,0 +1,12 @@
+package com.company.exceptions;
+
+public class CheckIndexException extends RuntimeException {
+
+    public CheckIndexException() {
+
```

Comment:
лучше всегда передавать краткое текстовое описание в конструктор родителя у исключений
```suggestion
super("Invalid index value");
```